docs/development/submitting-patches.rst

```diff
@@ -57,6 +57,67 @@ Important information should be documented with prose in the ``docs`` section.
 To ensure it builds and passes `doc8`_ style checks you can run
 ``make docs`` and ``make lint`` respectively.
 
+Keeping your local branch updated
+---------------------------------
+
+As you work, it is helpful to keep your local branch up-to-date with the PRs
+that are merged into the main Warehouse repository. Most people working on
+the project use this workflow.
+
+The following documentation assumes that you have git configured so that when
+you run the following command:
+
+.. code-block:: console
+
+  git remote -v
+
+Your output looks like this:
+
+.. code-block:: console
+
+  origin  https://github.com/USERNAME/warehouse.git (fetch)
+  origin  https://github.com/USERNAME/warehouse.git (push)
+  upstream  https://github.com/pypa/warehouse.git (fetch)
+  upstream  https://github.com/pypa/warehouse.git (push)
+
+
+In the example above, ``USERNAME`` is your username on GitHub.
+
+First, fetch the changes from the ``upstream`` Warehouse repository:
+
+.. code-block:: console
+
+  git fetch upstream
+
+Then, check out your master branch, and rebase the changes on top of it:
+
+.. code-block:: console
+
+  git checkout master
+  git rebase upstream/master
+
+Next, push the changes you have fetched to your ``origin`` branch on GitHub:
+
+.. code-block:: console
+
+  git checkout master
+  git push origin master
+
+Now your local branch and your ``origin`` are up-to-date with the most recent
+changes on the ``upstream`` Warehouse branch.
```

Comment:
I think you should use the word "master" more often here when referring to a master branch, to distinguish from remotes like "origin" & "upstream".